# HG changeset patch
# User Sean Farley <sean@mcs.anl.gov>
# Date 1332116384 18000
#      Sun Mar 18 19:19:44 2012 -0500
# Node ID b51a142f9789feb5bda1d1546b8ec8525f356f74
# Parent  91baffc7e83cf901cb335d75fdd8357d58548098
metis: rip out code copying of metis and use linker / header flags

diff -r 91baffc7e83c -r b51a142f9789 CMakeLists.txt
--- a/CMakeLists.txt	Sun Mar 18 19:19:44 2012 -0500
+++ b/CMakeLists.txt	Sun Mar 18 19:19:44 2012 -0500
@@ -2,14 +2,10 @@ cmake_minimum_required(VERSION 2.8)
 project(ParMETIS)
 
 set(GKLIB_PATH METIS/GKlib CACHE PATH "path to GKlib")
 set(METIS_PATH METIS CACHE PATH "path to METIS")
 
-# Symlink ./metis to wherever metis is. This allows files to be
-# included from metis/libmetis/.
-execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ${METIS_PATH} metis)
-
 # Search for MPI.
 # GK commented this out as it seems to be creating problems
 # include(FindMPI)
 # if(NOT MPI_FOUND)
 #   message(FATAL_ERROR "mpi is not found")
@@ -22,21 +18,24 @@ if(SHARED)
   set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/lib")
 else()
   set(ParMETIS_LIBRARY_TYPE STATIC)
 endif()
 
-include(${GKLIB_PATH}/GKlibSystem.cmake)
-
 # List of paths that the compiler will search for header files.
 # i.e., the -I equivalent
 include_directories(include)
 include_directories(${MPI_INCLUDE_PATH})
 include_directories(${GKLIB_PATH})
 include_directories(${METIS_PATH}/include)
+include_directories(${CMAKE_INSTALL_PREFIX}/include)
+
+# List of paths that the compiler will search for library files.
+# i.e., the -L equivalent
+link_directories(${METIS_PATH}/lib)
+link_directories(${CMAKE_INSTALL_PREFIX}/lib)
 
 # List of directories that cmake will look for CMakeLists.txt
-add_subdirectory(${METIS_PATH}/libmetis ${CMAKE_BINARY_DIR}/libmetis)
 add_subdirectory(include)
 add_subdirectory(libparmetis)
 add_subdirectory(programs)
 
 # This is for testing during development and is not being distributed
diff -r 91baffc7e83c -r b51a142f9789 Makefile
--- a/Makefile	Sun Mar 18 19:19:44 2012 -0500
+++ b/Makefile	Sun Mar 18 19:19:44 2012 -0500
@@ -22,11 +22,11 @@ systype = $(shell uname -s)
 BUILDDIR = build/$(systype)-$(cputype)
 
 # Process configuration options.
 CONFIG_FLAGS = -DCMAKE_VERBOSE_MAKEFILE=1
 ifeq ($(gklib_path), not-set)
-    gklib_path = metis/GKlib
+    gklib_path = headers
 endif
 ifeq ($(metis_path), not-set)
     metis_path = metis
 endif
 CONFIG_FLAGS += -DGKLIB_PATH=$(abspath $(gklib_path)) -DMETIS_PATH=$(abspath $(metis_path))
diff -r 91baffc7e83c -r b51a142f9789 libparmetis/parmetislib.h
--- a/libparmetis/parmetislib.h	Sun Mar 18 19:19:44 2012 -0500
+++ b/libparmetis/parmetislib.h	Sun Mar 18 19:19:44 2012 -0500
@@ -18,13 +18,13 @@
 
 #include <GKlib.h>
 
 #include <parmetis.h>
 
-#include "../metis/libmetis/gklib_defs.h"
+#include <gklib_defs.h>
 
-#include <mpi.h> 
+#include <mpi.h>
 
 #include <rename.h>
 #include <defs.h>
 #include <struct.h>
 #include <macros.h>
diff -r 91baffc7e83c -r b51a142f9789 programs/parmetisbin.h
--- a/programs/parmetisbin.h	Sun Mar 18 19:19:44 2012 -0500
+++ b/programs/parmetisbin.h	Sun Mar 18 19:19:44 2012 -0500
@@ -16,12 +16,12 @@
 #define DMALLOC			1
 */
 
 #include <GKlib.h>
 #include <parmetis.h>
+#include <gklib_defs.h>
 
-#include "../metis/libmetis/gklib_defs.h"
 #include "../libparmetis/rename.h"
 #include "../libparmetis/defs.h"
 #include "../libparmetis/struct.h"
 #include "../libparmetis/macros.h"
 #include "../libparmetis/proto.h"
